### DISCUSSION TITLE
Support parsing and production of "base-commit" and "prerequisite-patch-id" trailers in patches

### DISCUSSION THREAD

**Discussion by williewillus** [libgit2/libgit2#6440] [Ideas]
Original git documentation: https://git-scm.com/docs/git-format-patch#_base_tree_information

git format-patch has the ability to output in a patchset the upstream commit hash the patcheset is based on (the `base-commit` trailer), and any intermediate patchsets that are required by the stack (the `prerequisite-patch-id` trailers).

This allows users of the email patch-workflow and associated services (such as Sourcehut) to know where the patches have to be applied to test the given patches.

I took a brief look and I don't think this is supported by libgit2 at all at the moment.

**Comment by ethomson** [comment]
It's not but it would be reasonably straightforward to add this functionality. Pull requests for this would be welcome. 🙏

**Comment by williewillus** [comment]
Thanks! I'll log a formal issue to track implementation of this, and if
I have time I can try taking a look.
